crates/bitwarden-vault/src/cipher/cipher_client/create.rs

```diff
@@ -0,0 +1,387 @@
+use bitwarden_api_api::{apis::ciphers_api, models::CipherRequestModel};
+use bitwarden_core::{
+    key_management::{KeyIds, SymmetricKeyId},
+    require, ApiError, MissingFieldError, NotAuthenticatedError, OrganizationId, UserId,
+};
+use bitwarden_crypto::{
+    CompositeEncryptable, CryptoError, EncString, IdentifyKey, KeyStore, KeyStoreContext,
+    PrimitiveEncryptable,
+};
+use bitwarden_error::bitwarden_error;
+use bitwarden_state::repository::{Repository, RepositoryError};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+#[cfg(feature = "wasm")]
+use tsify::Tsify;
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+use super::CiphersClient;
+use crate::{
+    cipher_view_type::{CipherViewType, CipherViewTypeExt},
+    Cipher, CipherRepromptType, CipherType, CipherView, FieldView, FolderId, VaultParseError,
+};
+
+#[allow(missing_docs)]
+#[bitwarden_error(flat)]
+#[derive(Debug, Error)]
+pub enum CreateCipherError {
+    #[error(transparent)]
+    Crypto(#[from] CryptoError),
+    #[error(transparent)]
+    Api(#[from] ApiError),
+    #[error(transparent)]
+    VaultParse(#[from] VaultParseError),
+    #[error(transparent)]
+    MissingField(#[from] MissingFieldError),
+    #[error(transparent)]
+    NotAuthenticated(#[from] NotAuthenticatedError),
+    #[error(transparent)]
+    Repository(#[from] RepositoryError),
+}
+
+/// Request to add a cipher.
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
+#[cfg_attr(feature = "wasm", derive(Tsify), tsify(into_wasm_abi, from_wasm_abi))]
+pub struct CipherCreateRequest {
+    /// The ID of the user that is encrypting the cipher - this should always match the user
+    /// calling the API.
+    pub organization_id: Option<OrganizationId>,
+    pub folder_id: Option<FolderId>,
+    pub name: String,
+    pub notes: Option<String>,
+    pub r#type: CipherType,
+    pub favorite: bool,
+    pub reprompt: CipherRepromptType,
+    pub type_data: Option<CipherViewType>,
+    pub fields: Vec<FieldView>,
+    pub key: Option<EncString>,
+}
+
+impl CipherCreateRequest {
+    /// Generate a new key for the cipher, re-encrypting internal data, if necessary, and stores the
+    /// encrypted key to the cipher data.
+    pub fn generate_cipher_key(
```

Comment:
```suggestion
   fn generate_cipher_key(
```

I think this would be internal to this create?